File: UberMonkFish/nz/artemonkfish/misc/Strategy.java
package nz.artemonkfish.misc;

import nz.artemonkfish.ArteMonkFish;

/**
 * Created by IntelliJ IDEA.
 * User: Taylor
 * Date: 2/20/11
 * Time: 11:51 AM
 * Package: nz.uberfalconry;
 */
public abstract class Strategy extends Common
{
    public Strategy(ArteMonkFish parent) {
        super(parent);
    }

    public abstract void execute();

    public abstract boolean isValid();

    public abstract String getStatus();
}

File: ArteMiner/nz/arteminer/misc/Strategy.scala
package nz.arteminer.misc

/**
 * Created by IntelliJ IDEA.
 * User: Taylor
 * Date: 4/28/11
 * Time: 3:04 AM
 * Package: nz.arteminer.threads; */

trait Strategy {
  def getStatus:String
  def execute()
  def isValid:Boolean
}